### PR TITLE
Validate sub-identifier when encoding ObjectIdentifier

### LINKED ISSUE #97419: AsnWriter.WriteObjectIdentifier permits invalid sub identifier
### Description

"ITU-T X.680 | ISO/IEC 8824-1 requires that there be only three arcs allocated from the root node (with primary integer values of 0, 1, and 2), and at most forty arcs from the first two of these arcs (with primary integer values of 0 to 39)."

However, `WriteObjectIdentifier` does not validate this. This results in non-roundtrippable encoding.

Given that `WriteObjectIdentifier` goes through the trouble of doing other validations on the input, this seems like an oversight in validating when the root arc is 0 or 1.

### Reproduction Steps

Consider

```C#
AsnWriter asnWriter = new(AsnEncodingRules.DER);
asnWriter.WriteObjectIdentifier("1.255.3.0.9");
AsnReader reader = new (asnWriter.Encode(), AsnEncodingRules.DER);
Console.WriteLine(reader.ReadObjectIdentifier());
```

This will print out "2.215.3.0.9".

### Expected behavior

Throw "System.ArgumentException: The OID value is invalid. (Parameter 'oidValue')" like when the root arc is invalid.

### Actual behavior

Encodes incorrectly with no error.

### Regression?

Not a regression.

### Known Workarounds

_No response_

### Configuration

Impacts all versions of System.Formats.Asn1.

### Other information

_No response_

### PR DESCRIPTION
**Pull request by vcsjones**
Fixes #97419

### CODE REVIEW

File: src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.Oid.cs [hunk 1]
```diff
@@ -117,6 +117,12 @@ private void WriteObjectIdentifierCore(Asn1Tag tag, ReadOnlySpan<char> oidValue)
                 ReadOnlySpan<char> remaining = oidValue.Slice(2);
 
                 BigInteger subIdentifier = ParseSubIdentifier(ref remaining);
+
+                if (firstComponent <= 1 && subIdentifier >= 40)
```
Comment (vcsjones) [inline src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.Oid.cs:121]: I think the comment a few lines up (right before the diff cuts off) explains why this is here.

File: src/libraries/System.Formats.Asn1/tests/Writer/WriteObjectIdentifier.cs [hunk 2]
```diff
@@ -229,6 +229,8 @@ public void VerifyWriteObjectIdentifier_ConstructedIgnored(AsnEncodingRules rule
                 new object[] { "Invalid character second position", AsnEncodingRules.CER, "0,1.23" },
                 new object[] { "Invalid character second rid", AsnEncodingRules.DER, "0.1q.23" },
                 new object[] { "Invalid character third rid", AsnEncodingRules.BER, "0.1.23q" },
+                new object[] { "Invalid secord RID for first arc 0", AsnEncodingRules.DER, "0.40.1.2.3" },
+                new object[] { "Invalid secord RID for first arc 1", AsnEncodingRules.BER, "1.40.1.2.3" },
```
Comment (vcsjones) [inline src/libraries/System.Formats.Asn1/tests/Writer/WriteObjectIdentifier.cs:233]: Positive test already exists.

https://github.com/dotnet/runtime/blob/8575df9d294a25db06fdd138ca0ae5361a8ce5e2/src/libraries/System.Formats.Asn1/tests/Writer/WriteObjectIdentifier.cs#L189

File: src/libraries/System.Formats.Asn1/tests/Writer/WriteObjectIdentifier.cs [hunk 3]
```diff
@@ -229,6 +229,8 @@ public void VerifyWriteObjectIdentifier_ConstructedIgnored(AsnEncodingRules rule
                 new object[] { "Invalid character second position", AsnEncodingRules.CER, "0,1.23" },
                 new object[] { "Invalid character second rid", AsnEncodingRules.DER, "0.1q.23" },
                 new object[] { "Invalid character third rid", AsnEncodingRules.BER, "0.1.23q" },
+                new object[] { "Invalid second RID for first arc 0", AsnEncodingRules.DER, "0.40.1.2.3" },
```
Comment (vcsjones) [inline src/libraries/System.Formats.Asn1/tests/Writer/WriteObjectIdentifier.cs:232]: Positive test already exists for arc 2.

https://github.com/dotnet/runtime/blob/8575df9d294a25db06fdd138ca0ae5361a8ce5e2/src/libraries/System.Formats.Asn1/tests/Writer/WriteObjectIdentifier.cs#L189

**Review by bartonjs — APPROVED** [review]

**Comment by vcsjones** [comment]
Failures are known. Merging.

**Pull request merged**